pkg/kapp/app/recorded_app.go

```diff
@@ -140,17 +165,25 @@ func (a *RecordedApp) mergeAppUpdates(cm *corev1.ConfigMap, labels map[string]st
 }
 
 func (a *RecordedApp) Exists() (bool, string, error) {
-	_, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
-	if err != nil {
+	_, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.fqName, metav1.GetOptions{})
+	if err == nil {
+		return true, "", nil
+	}
+
+	if errors.IsNotFound(err) {
+		_, err = a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
```

Comment:
`Exists()` is always called before `Delete()` and `Rename()`, so should we save this information (that the app is new or old) somewhere and use it in `Delete()` and `Rename()` to avoid one additional call?

Reply:
I like that idea! Let me explore if that would be viable!

Reply:
Nice suggestion :) I ended up doing this, It reduces the calls by a bit! Thanks 🎉